### PR TITLE
add setup.py, workflows, unit tests, and examples

### PR DESCRIPTION
**Pull request by kta-intel**
This PR adds:

- `setup.py` to enable installation of `openfl-contrib` extension on top of latest base `openfl`
-  `openfl_contrib` folder structure to mirror `openfl`, containing:
   - `openfl_contrib.interface.aggregation_functions` with a custom weighted aggregation function (custom implementation of OpenFL's weighted average)
   - `openfl_contrib.pipelines` with an SKC compression pipeline (borrowed directly from base OpenFL)
-  `openfl_contrib_workspace`  to demonstrate how to incorporate custom plan with new agg functions and compression pipelines. Will also serve as the workspace for Task Runner test
- `.github/workflows` to run lint and tests
- `tests` directory containing:
   - `github` for CI testing of Task Runner API with custom plan and `CustomWeightedAverage` and `SKCPipeline`
   - `openfl-contrib` for individual unit tests for  `CustomWeightedAverage` and `SKCPipeline`

### CODE REVIEW

File: openfl_contrib/interface/aggregation_functions/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright (C) 2020-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Aggregation functions package."""
+
+from .custom_weighted_average import CustomWeightedAverage
+
+__all__ = [
+    'CustomWeightedAverage'
+    ]
```
Comment (teoparvanov) [inline openfl_contrib/interface/aggregation_functions/__init__.py:10]: nit - the indentation looks a bit off here

File: tests/github/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Copyright (C) 2020-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Tests package."""
```
Comment (teoparvanov) [inline tests/github/__init__.py:3]: I remember seeing a similar package structure somewhere under `openfl`, but I don't understand very well why we need a separate `github` package (supposedly for tests running in the CI?). I suggest to simplify, and move the the `test_task_runner.py` module right under `tests/openfl_contrib`.

**Review by teoparvanov — APPROVED** [review]
Thanks for submitting PR no. 1, @kta-intel 😉 ! It's great that you have created some basic infrastructure for contributors to start building with. I have a couple of suggestions for making the repo structure a bit more intuitive - but otherwise I believe it's in pretty good shape already!

File: tests/github/__init__.py (hunk 2 above)
Comment (kta-intel) [reply tests/github/__init__.py:3]: Similar to my other reply. A lot of the structural choices was made to mirror openfl base repo. For the sake of this first PR, we'll keep it as is, but let's discuss how to adjust the structure (if needed)

**Pull request merged**